domains/client/domain-operator/src/fraud_proof.rs

```diff
@@ -151,24 +151,30 @@ where
         _local_receipt: &ExecutionReceiptFor<Block, CBlock>,
         mismatch_type: BundleMismatchType,
         bundle_index: u32,
-        _bad_receipt_hash: H256,
+        bad_receipt_hash: H256,
     ) -> Result<FraudProof<NumberFor<PCB>, PCB::Hash>, FraudProofError>
     where
         PCB: BlockT,
     {
         match mismatch_type {
             // TODO: Generate a proper proof once fields are in place
             BundleMismatchType::TrueInvalid(_invalid_type) => Ok(FraudProof::InvalidBundles(
-                InvalidBundlesFraudProof::ValidAsInvalid(ValidAsInvalidBundleEntryFraudProof::new(
+                InvalidBundlesFraudProof::TrueInvalid(TrueInvalidBundleEntryFraudProof::new(
+                    bad_receipt_hash,
                     domain_id,
                     bundle_index,
+                    0,
+                    vec![],
+                    ProofDataPerExpectedInvalidBundle::OutOfRangeTx,
```

Comment:
The invalid bundle type should be determined by `invalid_type`, if a type is unimplemented then it should be handled by a branch like `_ => unimplemented!()`

Reply:
Same as above.